### DISCUSSION TITLE
Trace the full message even when it is very long

### DISCUSSION THREAD

**Discussion by hzhua** [langfuse/langfuse#4680] [Ideas]
### Describe the feature or potential improvement

Currently, langfuse will truncated a long message due to size exceeding limit. For long context applications that messages could reach 1 million tokens. Is there any way for langfuse to trace long messages without truncation?

**Comment by jannikmaierhoefer** [comment]
Hi @hzhua, you can add long messages as text media files. The way this works is that Langfuse uploads media files (long text, images, audio, etc) separately and then links them to the prompt. Have a look at our docs on this here: https://langfuse.com/docs/tracing-features/multi-modality
Let me know if this would be an option for you!

**Reply by marcklingen** [reply]
Adding to @jannikmaierhoefer's response here, It would be interesting here to automatically do this step within the client-sdks

What kinds of payloads do you have that are in the millions of tokens?

**Reply by hzhua** [reply]
We use long-context models like Gemini to summarize long documents like financial report.
